spec/ddtrace/contrib/rails/rack_spec.rb

```diff
@@ -341,22 +466,75 @@ def internal_server_error
     end
   end
 
-  context 'with error rendering partial template' do
-    subject { get '/error_partial' }
+  context 'with an explicitly raised 404' do
+    subject { get '/explicitly_not_found' }
+
+    it 'does not mark span as error' do
+      is_expected.to be_not_found
+
+      expect(spans).to have_at_least(1).item
+      request_span = spans[0]
+
+      expect(request_span.name).to eq('rack.request')
+      expect(request_span.span_type).to eq('web')
+      expect(request_span.resource).to eq('TestController#explicitly_not_found')
+      expect(request_span.get_tag('http.url')).to eq('/explicitly_not_found')
+      expect(request_span.get_tag('http.method')).to eq('GET')
+      expect(request_span.get_tag('http.status_code')).to eq('404')
+
+      if rails_older_than_3_2
+        # Old Rails does not have ActionDispatch::ExceptionWrapper, thus lets the error bubble up.
+        expect(request_span).to have_error
+      else
```

Comment:
This naming inconsistency was already there in the minitest tests, but it's a bit weird that actually in older Rails the test that checks that "it does not mark span as error" actually checks it should have an error.

Perhaps we could break the to/to_not have_error into a separate smaller `it`s that then uses `if: ...` to toggle on or off depending on the rails version?

Reply:
Done!